### PR TITLE
fix have_db_column.with_options misspelled options

### PR DESCRIPTION
**Pull request by rodriggochaves**
Hi!! This is my first contribuition to `shoulda-matchers`. If you got any feedback for this PR, I'll be very happy to listen.

This PR looks to solve the bug reported here #1281. 

### Solution descripton 

When there is an argument this matcher doesn't expect, it should raise an `ArgumentError`. I created a new protected method `validate_options` to be responsible for this.

### CODE REVIEW

File: lib/shoulda/matchers/active_record/have_db_column_matcher.rb
Comment (mcmire) [inline lib/shoulda/matchers/active_record/have_db_column_matcher.rb:141]: Would you mind putting this at the top of the class? That way it "pops" out.

File: spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb
Comment (mcmire) [inline spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb:105]: What do you think about testing what the error message is going to be rather than what kind of error it is? When you do that you might see that error could be improved a bit. I believe that as `invalid_options` is an array, when it gets interpolated it will appear like an array in the message. Not sure if that's what you were going after or not.

File: lib/shoulda/matchers/active_record/have_db_column_matcher.rb
Comment (mcmire) [inline lib/shoulda/matchers/active_record/have_db_column_matcher.rb:145]: Just a little style thing: we don't tend to use modifiers or `unless` — would you mind formatting this as:

``` ruby
if invalid_options.any?
  raise ArgumentError, "Unknown option(s) '#{invalid_options}'"
end
```

(although see note below about possibly reformatting the message)

**Review by mcmire — COMMENTED** [review]
Hi @rodriggochaves — thank you for this! It's nice when PRs are small :) Just had a few comments below.

File: spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb
Comment (rodriggochaves) [reply spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb]: Ah! Good observation!! I just changed the test and improved the message raised. Let me know what did you think about  it?

File: lib/shoulda/matchers/active_record/have_db_column_matcher.rb
Comment (rodriggochaves) [reply lib/shoulda/matchers/active_record/have_db_column_matcher.rb]: I moved but I've a question about the visibility of this constant. Shoud I use something to hide like `#private_constant`?

Comment (mcmire) [reply lib/shoulda/matchers/active_record/have_db_column_matcher.rb]: Sure, that would make sense if you want to do that.

File: spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb
Comment (mcmire) [inline spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb:105]: Okay, so now that you're testing multiple options, it seems like we could improve this message a bit. What if you change the above to 

``` ruby
raise ArgumentError, "Unknown option(s): #{invalid_options.map(&:inspect).join(", ")}"
```

Then you should get something like:

``` ruby
expect {
  have_db_column(:salary).with_options(preccision: 5, primaryy: true)
}.to raise_error("Unknown option(s): :preccision, :primaryy")
```

**Review by mcmire — COMMENTED** [review]
One more little thing but otherwise I'm good with this!

File: spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb
Comment (rodriggochaves) [reply spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb]: Ok! I improved the message!

**Review by mcmire — APPROVED** [review]
Sorry for the delay on this. This looks good to me, so thank you! It sounds like we may want to push this to the version after the next immediate one, so I'm going to hold off on merging but this looks ready to go once we are ready to merge.

**Comment by VSPPedro** [comment]
Thank you very much, @rodriggochaves!

@mcmire, I noticed that this PR was selected to be part of the 4.5.0. So, I think we can merge it.

**Comment by mcmire** [comment]
@VSPPedro Ah, sorry! For some reason I thought we were inserting a patch release before 4.5.0. In that case I'll merge this.

**Pull request merged**
